mab2rec/visualization.py

```diff
@@ -445,7 +445,7 @@ def plot_personalization_heatmap(recommendation_results: Union[Dict[str, pd.Data
 
         # Calculate percentage of recommendations by item within each cluster
         df = rec_sorted_df.groupby(['cluster', item_id_col]).size()
-        df = df.groupby(level=0).apply(lambda x: x / float(x.sum())).reset_index()
+        df = df.groupby(level=0).apply(lambda x: x / float(x.sum())).droplevel(level=0).reset_index()
```

Comment:
This fails for Python 3.7 so not fully backward compatible with older Pandas versions (see CI logs). 
@nagireddyakshay can you see if there is a solution that works for both Py3.7 (pandas <2.0) and Py3.8+ (pandas 2.0+)?